docs/fundamentals/wadm/api.md

```diff
@@ -2,11 +2,444 @@
+we do not require that the version field be semver.
+
+### Delete Models
+
+`wadm.api.{lattice}.model.del.{name}`
+
+Deletes the named model with the supplied version number. If the `delete_all` flag is set, then all
+versions of this model will be deleted. If you delete a model version (or the whole model) that is
+currently deployed, then the resources will be undeployed automatically as well.
+
+#### Request
+
+```json
+{
+  "version": "1.0",
+  "delete_all": false,
+}
+```
+
+#### Response
+
+```json
+{
+  "result": "deleted|error|noop",
+  "undeploy": true,
+  "message": "Model version 1.0 deleted"
+}
+```
+
+The response indicates success, failure, or noop (i.e. nothing was deleted because it didn't exist),
+and if an `undeploy` operation occurred as part of the delete.
+
+## Managing Deployments
+
+Deployments are discrete instances of autonomous control agents that monitor an application model
+against the real-time, discovered state of a lattice. This agent monitors lattice events, compares
+lattice state against the desired state of the model, and issues lattice control interface commands
+in order to compensate for observed gaps.
+
+### Deploy
+
+`wadm.api.{lattice}.model.deploy.{name}`
+
+Deploys the indicated model. This operation is idempotent, and will simply "succeed" if a request to
+deploy an already deployed model is received. The previous version, if deployed, does not need to be
+_undeployed_. Instead, the desired state of the new version is compared against the observed state
+of the lattice and the autonomous agents will compensate accordingly. If the request payload is
+empty, or the `version` field contains the keyword `latest`, the newest model spec stored will be
+deployed.
+
+#### Request
+
+Empty body to deploy the latest version
+
+or
+
+```json
+{
+  "version": "version to deploy"
+}
+```
+
+**Response**:
+
+```json
+{
+  "result": "acknowledged|error|notfound",
+  "message": "..."
+}
+```
+
+It's important to note that the success/fail of this call doesn't indicate completed deployment,
+only success or failure of the receipt of the deployment request. In other words, the response is an
+_ack_. `notfound` is a specific error condition that indicates the model or version was not found
+
+### Undeploy
+
+`wadm.api.{lattice}.model.undeploy.{name}`
+
+Undeploys the currently deployed version of a model, destroying all resources managed by this model.
+If the model is already undeployed, the operation is idempotent and will return success along with
+an appropriate message.
+
+:::note
+The response is delivered immediately as an acknowledgement, as it can take several minutes
+to complete an undeploy.
+:::
+
+#### Request
+
+Empty body
+
+or
+
+```json
+{
+  "non_destructive": true
+}
+```
+
+for a non-destructive undeploy. This will stop managing, but will not destroy resources.
+
+:::warning
+This feature is currently not implemented, but will be added in a future version
```

Comment:
IMO we should hide this until it's implemented

At the very least, make the 3 warning messages consistent

Reply:
The warnings are different because there are different plans for each of them, I think I'll hide deployment history, but keep status around as we _are_ planning on implementing that

Reply:
I removed some stuff and clarified others. Let me know what you think